sphinx/environment/__init__.py

```diff
@@ -760,11 +709,210 @@ def check_consistency(self) -> None:
                                location=docname)
 
         # call check-consistency for all extensions
-        for domain in self.domains.values():
-            domain.check_consistency()
+        self.domains._check_consistency()
         self.events.emit('env-check-consistency', self)
 
 
+class _DomainsType:
+    __slots__ = (
+        '_data',
+        'c_domain',
+        'changeset_domain',
+        'citation_domain',
+        'cpp_domain',
+        'index_domain',
+        'javascript_domain',
+        'math_domain',
+        'python_domain',
+        'restructuredtext_domain',
+        'standard_domain',
+    )
+
+    #: First-party domains in :mod:`sphinx.domains`
+    _core_domains: Final = frozenset({
+        'std', 'c', 'cpp', 'js', 'py', 'rst',
+        'changeset', 'citation', 'index', 'math',
+    })  # fmt: skip
```

Comment:
```suggestion
    })
```
I would use the default formatting here

Reply:
This isn't actually auto-formatted yet. But the default formatting would take up even more vertical space?

Reply:
Yes it would. Is that a problem?

Another benefit is that for future additions to the predefined types you can 'git blame' to see when it was added